src/main/java/com/gravity9/mongocdc/MongoChangeStreamWorker.java

```diff
@@ -71,6 +74,14 @@ public void stop() {
         log.info("Worker for partition {} on collection {} stopped!", partition, mongoConfig.getCollectionName());
     }
 
+    public void awaitInitialization() {
+        try {
+            initializationLatch.await();
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
Is this a proper exception to be thrown? How should the application behave when awaiting is interrupted?

Reply:
Application needs to catch StartFailureException (added just now).